crates/sdk/src/network-v2/client.rs

```diff
@@ -0,0 +1,228 @@
+        &self,
+        request_id: &[u8],
+    ) -> Result<(GetProofRequestStatusResponse, Option<P>)> {
+        let mut rpc = self.get_rpc().await?;
+
+        let res = rpc
+            .get_proof_request_status(GetProofRequestStatusRequest {
+                request_id: request_id.to_vec(),
+            })
+            .await?
+            .into_inner();
+        let status = ProofStatus::try_from(res.status)?;
+        let proof = match status {
+            ProofStatus::Fulfilled => {
+                log::info!("Proof request fulfilled");
+                let proof_uri = res.proof_uri.as_ref().expect("no proof url");
+                let proof_bytes = self.download_artifact(proof_uri).await?;
+                Some(bincode::deserialize(&proof_bytes).context("Failed to deserialize proof")?)
+            }
+            _ => None,
+        };
+
+        Ok((res, proof))
+    }
+
+    /// Get all the proof requests for a given status. Also filter by version if provided.
+    pub async fn get_filtered_proof_requests(
+        &self,
+        status: ProofStatus,
+        version: Option<&str>,
+    ) -> Result<GetFilteredProofRequestsResponse> {
+        let mut rpc = self.get_rpc().await?;
+        let res = rpc
+            .get_filtered_proof_requests(GetFilteredProofRequestsRequest {
+                status: status.into(),
+                version: version.map(|v| v.to_string()).unwrap_or_default(),
+            })
+            .await?
+            .into_inner();
+
+        Ok(res)
+    }
+
+    /// Creates a proof request with the given ELF and stdin.
+    #[allow(clippy::too_many_arguments)]
+    pub async fn request_proof(
+        &self,
+        elf: &[u8],
+        stdin: &SP1Stdin,
+        vk: &SP1VerifyingKey,
+        mode: ProofMode,
+        _version: &str,
+        strategy: ProofStrategy,
+        timeout_secs: u64,
+        cycle_limit: u64,
+    ) -> Result<RequestProofResponse> {
+        // Calculate the deadline.
+        let start = SystemTime::now();
+        let since_the_epoch = start.duration_since(UNIX_EPOCH).expect("Invalid start time");
+        let deadline = since_the_epoch.as_secs() + timeout_secs;
+
+        // Create the program and stdin artifacts.
+        let mut store = self.get_store().await?;
+        let mut store_clone = store.clone();
+        let program_promise = self.create_artifact_with_content(&mut store, &elf);
+        let stdin_promise = self.create_artifact_with_content(&mut store_clone, &stdin);
+        let (program_uri, stdin_uri) = try_join!(program_promise, stdin_promise)?;
+
+        // Serialize the vkey.
+        let vkey = bincode::serialize(&vk)?;
+
+        // Send the request.
+        let mut rpc = self.get_rpc().await?;
+        let nonce = self.get_nonce().await?;
+        let request_body = RequestProofRequestBody {
+            nonce,
+            // version: version.to_string(),
+            version: "sp1-v3.0.0-rc1".to_string(),
```

Comment:
can this be format!("sp1-{}", SP1_VERSION) which is in lib.rs for core/machine

Reply:
fixed